src/pages/index.js

```diff
@@ -73,57 +68,195 @@ function handleImageClick(cardData) {
 const cardSection = new Section(
   {
     items: initialCards,
-    renderer: renderCard,
+    renderer: (item) => {
+      const cardELement = createCard(item);
+      cardSection.addItem(cardELement);
+    },
   },
-  cardListEl
+  ".cards__list"
 );
 
-cardSection.renderItems();
+//Event Listeners
 
-// Render Cards
-function renderCard(item, method = "addItem") {
-  const cardElement = createCard(item);
-  cardSection[method](cardElement);
-}
+addNewCardButton.addEventListener("click", () => {
+  addCardFormPopup.open();
+});
+
+profileEditButton.addEventListener("click", () => {
+  // editProfileFormValidator.resetValidation();
+  const inputData = userInfo.getUserInfo();
+  // console.log(inputData);
+  profileNameInput.value = inputData.name;
+  profileDescriptionInput.value = inputData.about;
+  profileEditPopup.open();
+});
+
+addCardFormValidator.enableValidation();
+editProfileFormValidator.enableValidation();
 
 //Functions
 function createCard(cardData) {
-  const card = new Card(cardData, "#card-template", handleImageClick);
-  return card.getView();
+  const card = new Card(
+    cardData,
+    "#card-template",
+    handleImageClick,
+    handleDeleteCardSubmit,
+    likeCard,
+    unlikeCard
+  );
+  return card.generateCard();
 }
 
 function handleProfileEditSubmit(inputData) {
-  userInfo.setUserInfo({
-    name: inputData.name,
-    job: inputData.description,
-  });
+  profileEditPopup.renderLoading(true);
+
+  api
+    .setUserInfo(inputData.name, inputData.description)
+    .then(() => {
+      userInfo.setUserInfo({
+        name: inputData.name,
+        about: inputData.description,
+      });
+      profileEditPopup.close();
+    })
+    .catch((err) => {
+      console.error("Error updating user info", err);
+      alert(err);
+    })
+    .finally(() => {
+      console.log("Edit form complete");
+      profileEditPopup.renderLoading(false);
+    });
 
-  profileEditPopup.close();
+  // profileEditPopup.close();
 }
 
 function handleAddCardSubmit(inputData) {
-  const cardData = {
-    name: inputData.title,
-    link: inputData.url,
-  };
-  renderCard(cardData);
-  addCardFormValidator.disableButton();
-  addCardFormPopup.close();
-  addCardForm.reset();
+  const name = inputData.title;
+  const link = inputData.url;
+
+  addCardFormPopup.renderLoading(true);
+  api
+    .uploadCard({ name, link })
+    .then((cardData) => {
+      const card = createCard(cardData);
+      // cardAddForm.setLoading(false);
+      cardSection.addItem(card);
+      addCardFormPopup.close();
+      addCardForm.reset();
```

Comment:
* You need to **`disable`** the submit button in the `card form` if the inputs are **empty**. It’s needed because I can click the button and add an empty card, but it should **not** happen. After the first submitting the button gets activated, and you need to **deactivate** it again. Even if I can't add an empty card, the button should look **disabled** (gray) anyway and have attribute `disabled`.

Reply:
Hi! I checked my submit button and it's disabled when it has to be disabled and It don't let me add any empty cards. Maybe I'm missing something that I don't see, can you be more specific please. I really appreciate your comments by the way, thankyou!

Reply:
> Hi! I checked my submit button and it's disabled when it has to be disabled and It don't let me add any empty cards. Maybe I'm missing something that I don't see, can you be more specific please. I really appreciate your comments by the way, thankyou!

Hi)

Have a look here. The button is not disabled (and not gray) after submission
![image](https://github.com/user-attachments/assets/819377ac-7b63-44c3-b8ab-111026be46bb)